core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1217,9 +1207,12 @@ public void run() {
                     return;
                 }
                 PlayerStatus playerStatus = mediaPlayer.getPlayerStatus();
-                PlaybackServiceNotificationBuilder notificationBuilder =
-                        new PlaybackServiceNotificationBuilder(PlaybackService.this);
-                notificationBuilder.addMetadata(playable, mediaSession.getSessionToken(), playerStatus, isCasting);
+                notificationBuilder = new PlaybackServiceNotificationBuilder(PlaybackService.this);
+                notificationBuilder.setMetadata(playable, mediaSession.getSessionToken(), playerStatus, isCasting);
+
+                if (Build.VERSION.SDK_INT < 29) {
+                    notificationBuilder.updatePosition(getCurrentPosition(), getCurrentPlaybackSpeed());
```

Comment:
The line (intermittently) threw IllegalStateException due to Exo-player accessed on the wrong thread. I'll attach the details in a separate comment.